operators/src/processing/temporal_raster_aggregation.rs

```diff
@@ -0,0 +1,1074 @@
+use crate::{
+    adapters::{RasterOverlapAdapter, SubQueryTileAggregator},
+    engine::{
+        InitializedOperator, InitializedRasterOperator, QueryRectangle, RasterQueryProcessor,
+        RasterResultDescriptor, TypedRasterQueryProcessor,
+    },
+    error,
+    util::Result,
+};
+use futures::{Future, FutureExt, StreamExt, TryFuture};
+use geoengine_datatypes::raster::GridOrEmpty;
+use geoengine_datatypes::{
+    primitives::{SpatialBounded, TimeInstance, TimeInterval, TimeStep},
+    raster::{Grid2D, Pixel, RasterTile2D, TileInformation, TilingSpecification},
+};
+use serde::{Deserialize, Serialize};
+use snafu::ensure;
+use typetag;
+
+use crate::engine::{ExecutionContext, Operator, RasterOperator, SingleRasterSource};
+
+#[derive(Debug, Clone, Copy, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TemporalRasterAggregationParameters {
+    aggregation_type: AggregationType,
+    window: TimeStep,
```

Comment:
window_size?

Reply:
window is a TimeStep which is Granularity + step_size